### ISSUE TITLE
Building Moya using Carthage

### ISSUE THREAD

**Issue by mobilestar0223** [Moya/Moya#756]
I have encountered below error when I updated Moya in Carthage.
I have found same issues and tried all suggestion. These didn't work for me.
Please help me!

```
The following build commands failed:
	CompileSwift normal arm64 /Volumes/Data/MyProject/Carthage/Checkouts/Moya/Source/Error.swift
	CompileSwift normal arm64 /Volumes/Data/MyProject/Carthage/Checkouts/Moya/Source/RxSwift/Moya+RxSwift.swift
	CompileSwift normal arm64 /Volumes/Data/MyProject/Carthage/Checkouts/Moya/Source/Image.swift
	CompileSwift normal arm64 /Volumes/Data/MyProject/Carthage/Checkouts/Moya/Source/Plugins/NetworkActivityPlugin.swift
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(5 failures)
```

**Comment by fruitcoder** [comment]
Carthage Version: 0.18.1
xcrun swift -version: Apple Swift version 3.0.1 (swiftlang-800.0.58.6 clang-800.0.42.1)
Xcode: Xcode 8.1 8B62

I also have an issue when resolving Moya dependencies in the following line
`RxCococa*** Building scheme "RxCocoa-iOS" in Rx.xcworkspace`

after I tried to rebuild the frameworks with `carthage update --platform iOS --no-use-binaries` (for the new Swift 3.0.1). Should this be posted in RxSwift or might this be resolved by updating the Cartfile in Moya?
The specific error in xcodebuilt output is `duplicate symbol __TMRbBp` in `...KVOObservable.o` and `...UIButton+Rx.o` which might be related to [this issue](https://github.com/ReactiveX/RxSwift/issues/970)

**Comment by Tynox** [comment]
I've got the same error of "RxCocoa Building error".

**Comment by AndrewSB** [comment]
Does it only come up when building RxCocoa through Moya? Not if you just have a Cartfile with `github "ReactiveX/RxSwift" ~> 3.0`?

If the issue is in fact with RxSwift, then it should be tracked on their library, on the issue @fruitcoder  mentioned https://github.com/ReactiveX/RxSwift/issues/970.

**Comment by mobilestar0223** [comment]
Hello, Andrew.
Thanks for your advice.
But it doesn't work for me also. I updated RxSwift from master repo again and also update Moya.
But same error occurred.
I don't know this is because RxSwift or Moya.
Thanks.

**Comment by Tynox** [comment]
@AndrewSB   I just tested a new project which has a Cartfile with only `github "ReactiveX/RxSwift" ~> 3.0`. Everything works fine.  And I tested a new project with only 'github "Moya/Moya" "8.0.0-beta.2"'. I got this error:

`The following build commands failed:
	Ld /Users/Alan/Library/Application\ Support/io.schwa.Punic/DerivedData/Build/Intermediates/Rx.build/Release-iphoneos/RxCocoa-iOS.build/Objects-normal/armv7/RxCocoa normal armv7
	Ld /Users/Alan/Library/Application\ Support/io.schwa.Punic/DerivedData/Build/Intermediates/Rx.build/Release-iphoneos/RxCocoa-iOS.build/Objects-normal/arm64/RxCocoa normal arm64
	CreateUniversalBinary /Users/Alan/Library/Application\ Support/io.schwa.Punic/DerivedData/Build/Products/Release-iphoneos/RxCocoa.framework/RxCocoa normal armv7\ arm64
	GenerateDSYMFile /Users/Alan/Library/Application\ Support/io.schwa.Punic/DerivedData/Build/Products/Release-iphoneos/RxCocoa.framework.dSYM /Users/Alan/Library/Application\ Support/io.schwa.Punic/DerivedData/Build/Products/Release-iphoneos/RxCocoa.framework/RxCocoa
(4 failures)`

**Comment by AndrewSB** [comment]
Can you delete your derived data and try again?

**Comment by Tynox** [comment]
Cleared derived data. Still got this error.

**Comment by AndrewSB** [comment]
Hmm, so it's working on me when I check specify `github "Moya/Moya" "master"`. Testing now with the `8.0.0-beta.2` release.

`master` output:

```
➜  how-carthage-works carthage update --platform iOS
*** Fetching Moya
*** Fetching RxSwift
*** Fetching Alamofire
*** Fetching Result
*** Checking out Alamofire at "4.0.1"
*** Downloading Result.framework binary at "3.0.0"
*** Checking out ReactiveSwift at "1.0.0-alpha.3"
*** Checking out RxSwift at "3.0.1"
*** Checking out Moya at "1b1edcf782bdf9eea0fa0c55c18275904d35d1e6"
*** xcodebuild output can be found in /var/folders/6y/2zb68tcs4nn6v899_jz0nmfr0000gn/T/carthage-xcodebuild.nCphgo.log
*** Building scheme "Alamofire iOS" in Alamofire.xcworkspace
*** Building scheme "ReactiveSwift-iOS" in ReactiveSwift.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/asb/how-carthage-works/Carthage/Checkouts/ReactiveSwift/ReactiveSwift.playground' in workspace '/Users/asb/how-carthage-works/Carthage/Checkouts/ReactiveSwift/ReactiveSwift.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/asb/how-carthage-works/Carthage/Checkouts/ReactiveSwift/ReactiveSwift.playground' in workspace '/Users/asb/how-carthage-works/Carthage/Checkouts/ReactiveSwift/ReactiveSwift.xcworkspace'.
*** Building scheme "RxBlocking-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "RxTests-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "RxCocoa-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "RxSwift-iOS" in Rx.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
--- xcodebuild: WARNING: Unable to open project file '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/asb/how-carthage-works/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
*** Building scheme "ReactiveMoya iOS" in Moya.xcodeproj
*** Building scheme "RxMoya iOS" in Moya.xcodeproj
*** Building scheme "Moya iOS" in Moya.xcodeproj
```

**Comment by mobilestar0223** [comment]
I have deleted whole derived data.
Also checkout Moya from master.
Still get same error.

**Comment by AndrewSB** [comment]
Hmm, I'm definitely seeing the issue on the latest release, but not on `master`. It seems to come and go.

@mobilestar0223 after it fails, can you open the `Rx.xcodeproj` inside `Carthage/Checkouts/RxSwift` and see if you can build the RxCocoa target from Xcode? If not, can you paste your error?
Trying to get a handle on what the issue is.

We may just want to do a new release soon -- not sure if that will fix the problem though, let's investigate it

**Comment by AndrewSB** [comment]
I think I've got a handle on the issue - I'm only seeing the `Ld` error from `RxCocoa` if I'm trying to build version `3.0.0-beta.1`, if I checkout `RxSwift` at `3.0.1` instead, I don't get the build failure

So we should release master as a new version ASAP, since our latest release (https://github.com/Moya/Moya/releases/tag/8.0.0-beta.3) points to RxSwift `3.0.0-beta.1`.

@ashfurrow can we do a new release?

**Comment by ashfurrow** [comment]
Aye, I'll do a release asap. Thanks for looking into this!

**Comment by ashfurrow** [comment]
All done 👍 Released as 8.0.0-beta.4, us know if this works. I've opened https://github.com/Moya/Moya/issues/759 to decide if we want to release that as a 8.0.0 final. Let

**Comment by AndrewSB** [comment]
Fix for this issue: target Moya `8.0.0-beta.4`.

If anyone is having issues, please reopen this. Im happy to help!
